### PR TITLE
Implement about us create team member

### PR DESCRIPTION
**Pull request by alexkocharyan01**

### CODE REVIEW

File: src/components/AboutUsAddMember/AboutUsAddMember.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import { useImageUpload } from '../../hooks/ImageUpload'
+
+const defaultValues = {
+  header: '',
+  paragraph: '',
+  imageDescription: '',
+  imageURL: null,
+}
+const AboutUsAddMember = () => {
+  const { imageLoaded, handleFileChange } = useImageUpload()
```
Comment (ArthurO2002) [inline src/components/AboutUsAddMember/AboutUsAddMember.tsx:24]: File should be AboutUs/AddMember/AboutUsMember.tsx

File: src/manager/aboutUs/aboutUsManager.ts [hunk 2]
```diff
@@ -12,6 +12,19 @@ const aboutUsManager = {
       `/core/v1/${baseURL}`
     )
   },
+  createTeamMember(formData: ICreateTeamMember) {
+    return axiosInstance.post<ICreateTeamMember, IAxiosResponse<ICreateTeamMember>>(
```
Comment (ArthurO2002) [inline src/manager/aboutUs/aboutUsManager.ts:16]:
```suggestion
  createTeamMember(params: ICreateTeamMember) {
    return axiosInstance.post<ICreateTeamMember, IAxiosResponse<ICreateTeamMember>>(
```

File: src/pages/about-us/index.tsx [hunk 3]
```diff
@@ -1,22 +1,41 @@
...
 
   const [t] = useTranslation()
 
   useEffect(() => {
     dispatch(aboutUsMiddleware.getAboutUsList())
   }, [])
 
+  useEffect(() => {
+    if (isTeamMemberSubmitSuccess) {
+      dispatch(aboutUsMiddleware.resetCreateTeamMemberSubmitSuccess())
```
Comment (ArthurO2002) [inline src/pages/about-us/index.tsx:34]: Why it is needed?

File: src/validation/aboutUs/aboutUs.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+export const createAboutUsValidationSchema = object().shape({
+  header: string().required('header is requier'),
+  paragraph: string().required('paragraph is requier'),
+  imageDescription: string().required('paragraph is requier'),
+  imageURL: mixed()
+    .test('required', 'You need to provide a file', (file) => {
+      if (file) {
+        return true
+      }
```
Comment (ArthurO2002) [inline src/validation/aboutUs/aboutUs.ts:11]: Can we test also file type?

File: src/components/AboutUsAddMember/AboutUsAddMember.tsx (hunk 1 above)
Comment (alexkocharyan01) [reply src/components/AboutUsAddMember/AboutUsAddMember.tsx]: done

File: src/manager/aboutUs/aboutUsManager.ts (hunk 2 above)
Comment (alexkocharyan01) [reply src/manager/aboutUs/aboutUsManager.ts]: done

File: src/pages/about-us/index.tsx (hunk 3 above)
Comment (alexkocharyan01) [reply src/pages/about-us/index.tsx]: i remove it

File: src/components/AboutUs/AddMember/AboutUsMember.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            <div className="space-y-4">
+              <div>
+                {imageLoaded ? (
+                  <Image
+                    src={imageLoaded}
+                    loader={() => imageLoaded}
+                    width={500}
+                    height={680}
+                    alt="picture"
+                    className="h-[345px] w-[500px]"
```
Comment (ArthurO2002) [inline src/components/AboutUs/AddMember/AboutUsMember.tsx:80]: Should be same
add class to contain

File: src/components/AboutUs/AddMember/AboutUsMember.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                  />
+                ) : null}
+              </div>
+              <div
+                className={`${
+                  imageLoaded !== undefined ? 'flex flex-col-reverse' : 'flex w-full justify-center'
+                }`}
```
Comment (ArthurO2002) [inline src/components/AboutUs/AddMember/AboutUsMember.tsx:87]: imageLoaded
```suggestion
                  !imageLoaded ? 'flex flex-col-reverse' : 'flex w-full justify-center'
                }`}
```

File: src/hooks/ImageUpload/ImageUpload.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React, { useState } from 'react'
+
+export const useImageUpload = () => {
+  const [imageLoaded, setImageLoaded] = useState<string>()
+
```
Comment (ArthurO2002) [inline src/hooks/ImageUpload/ImageUpload.tsx:5]: Should have initial state for example null

File: src/components/AboutUs/AddMember/AboutUsMember.tsx (hunk 6 above)
Comment (ArthurO2002) [reply src/components/AboutUs/AddMember/AboutUsMember.tsx:87]: We should never have undefined in our defined variables

File: src/redux/slices/aboutUs/middleware.ts [hunk 8]
```diff
@@ -22,6 +23,26 @@ const getAboutUsList = () => async (dispatch: AppDispatch) => {
...
 
+const createTeamMember = (formData: ICreateTeamMember) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setCreateTeamMemberSubmitSuccess(true))
+
+    await API.aboutUs.createTeamMember(formData)
+
+    dispatch(getAboutUsList())
+
+    await dispatch(getAboutUsList())
```
Comment (ArthurO2002) [inline src/redux/slices/aboutUs/middleware.ts:34]:
```suggestion
    dispatch(aboutUsMiddleware.getAboutUsList())
```

File: src/ui-components/FileField/FileField.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  id?: string
+  inputRef?: RefObject<HTMLInputElement>
+  handleFileChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
+}
+
+const FileField: FC<ITextFieldProps> = ({ fieldName, inputRef, handleFileChange }) => {
+  const { field, fieldState } = useController({ name: fieldName })
+
+  return (
+    <div className={`${fieldState.error && 'text-blue'}`}>
```
Comment (ArthurO2002) [inline src/ui-components/FileField/FileField.tsx:20]: use primary color instead

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/hooks/ImageUpload/ImageUpload.tsx (hunk 7 above)
Comment (alexkocharyan01) [reply src/hooks/ImageUpload/ImageUpload.tsx]: done

File: src/components/AboutUs/AddMember/AboutUsMember.tsx (hunk 6 above)
Comment (alexkocharyan01) [reply src/components/AboutUs/AddMember/AboutUsMember.tsx]: done

File: src/redux/slices/aboutUs/middleware.ts (hunk 8 above)
Comment (alexkocharyan01) [reply src/redux/slices/aboutUs/middleware.ts]: done

File: src/ui-components/FileField/FileField.tsx (hunk 9 above)
Comment (alexkocharyan01) [reply src/ui-components/FileField/FileField.tsx]: done

File: src/components/AboutUs/AddMember/AboutUsMember.tsx (hunk 5 above)
Comment (alexkocharyan01) [reply src/components/AboutUs/AddMember/AboutUsMember.tsx]: done

File: src/validation/aboutUs/aboutUs.ts (hunk 4 above)
Comment (alexkocharyan01) [reply src/validation/aboutUs/aboutUs.ts]: done

File: src/manager/aboutUs/aboutUsManager.ts [hunk 10]
```diff
@@ -12,6 +12,19 @@ const aboutUsManager = {
       `/core/v1/${baseURL}`
     )
   },
+  createTeamMember(params: ICreateTeamMember) {
+    return axiosInstance.post<ICreateTeamMember, IAxiosResponse<ICreateTeamMember>>(
+      `/core/v1/${baseURL}`,
+      {
+        ...params,
+      },
+      {
```
Comment (ArthurO2002) [inline src/manager/aboutUs/aboutUsManager.ts:21]:
```suggestion
  createTeamMember(data: ICreateTeamMember) {
    return axiosInstance.post<ICreateTeamMember, IAxiosResponse<ICreateTeamMember>>(
      `/core/v1/${baseURL}`,
      data,
      {
```

File: src/ui-components/FileField/FileField.tsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    <div>
+      <ImageInput
+        {...field}
+        inputRef={inputRef}
+        type="file"
+        onChange={(event) => {
+          if (handleFileChange) {
+            handleFileChange(event)
+            field.onChange(event.target.files && event.target.files?.[0])
+          }
```
Comment (ArthurO2002) [inline src/ui-components/FileField/FileField.tsx:29]: Create seperate function

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/manager/aboutUs/aboutUsManager.ts (hunk 10 above)
Comment (alexkocharyan01) [reply src/manager/aboutUs/aboutUsManager.ts]: done

File: src/ui-components/FileField/FileField.tsx (hunk 11 above)
Comment (alexkocharyan01) [reply src/ui-components/FileField/FileField.tsx]: done

**Review by ArthurO2002 — APPROVED** [review]

**Pull request merged**